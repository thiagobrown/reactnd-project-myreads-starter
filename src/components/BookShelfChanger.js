import React from 'react'
import PropTypes from 'prop-types'

const BookShelfChanger = ({ book, onUpdateBook }) => {
  return (
    <div className="book-shelf-changer">
      <select value={book.shelf}
        onChange={(e) => onUpdateBook(book, e.target.value)}>
        <option value="" disabled>Move to...</option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        <option value="none">None</option>
      </select>
    </div>
  )
}

BookShelfChanger.propTypes = {
  book: PropTypes.object.isRequired,
  onUpdateBook: PropTypes.func.isRequired
}

export default BookShelfChanger;
