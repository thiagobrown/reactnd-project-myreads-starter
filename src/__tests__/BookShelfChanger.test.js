import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import BookShelfChanger from './../components/BookShelfChanger'

describe('Testing BookShelfChanger Component', () => {
  let props = {
    book: {},
    onUpdateBook: jest.fn()
  }

  const wrapper = shallow(<BookShelfChanger {...props} />);

  it('render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('check render when state changes', () => {
    wrapper.setProps({
      book: {
        id: 1,
        title: 'Book One',
        authors: ['Authors One'],
        imageLinks: { thumbnail: 'url/book/one' },
        shelf: 'read'
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});