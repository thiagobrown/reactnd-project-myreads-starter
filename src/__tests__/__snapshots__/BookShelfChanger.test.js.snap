exports[`Testing BookShelfChanger Component check render when state changes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookShelfChanger
    book={
        Object {
            "authors": Array [
              "Authors One",
            ],
            "id": 1,
            "imageLinks": Object {
              "thumbnail": "url/book/one",
            },
            "shelf": "read",
            "title": "Book One",
          }
    }
    onUpdateBook={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value="read">
        <option
                disabled={true}
                value="">
                Move to...
        </option>
        <option
                value="currentlyReading">
                Currently Reading
        </option>
        <option
                value="wantToRead">
                Want to Read
        </option>
        <option
                value="read">
                Read
        </option>
        <option
                value="none">
                None
        </option>
</select>,
      "className": "book-shelf-changer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            disabled={true}
            value="">
            Move to...
</option>,
          <option
            value="currentlyReading">
            Currently Reading
</option>,
          <option
            value="wantToRead">
            Want to Read
</option>,
          <option
            value="read">
            Read
</option>,
          <option
            value="none">
            None
</option>,
        ],
        "onChange": [Function],
        "value": "read",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Move to...",
            "disabled": true,
            "value": "",
          },
          "ref": null,
          "rendered": "Move to...",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Currently Reading",
            "value": "currentlyReading",
          },
          "ref": null,
          "rendered": "Currently Reading",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Want to Read",
            "value": "wantToRead",
          },
          "ref": null,
          "rendered": "Want to Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Read",
            "value": "read",
          },
          "ref": null,
          "rendered": "Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "None",
            "value": "none",
          },
          "ref": null,
          "rendered": "None",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value="read">
          <option
                    disabled={true}
                    value="">
                    Move to...
          </option>
          <option
                    value="currentlyReading">
                    Currently Reading
          </option>
          <option
                    value="wantToRead">
                    Want to Read
          </option>
          <option
                    value="read">
                    Read
          </option>
          <option
                    value="none">
                    None
          </option>
</select>,
        "className": "book-shelf-changer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              value="">
              Move to...
</option>,
            <option
              value="currentlyReading">
              Currently Reading
</option>,
            <option
              value="wantToRead">
              Want to Read
</option>,
            <option
              value="read">
              Read
</option>,
            <option
              value="none">
              None
</option>,
          ],
          "onChange": [Function],
          "value": "read",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Move to...",
              "disabled": true,
              "value": "",
            },
            "ref": null,
            "rendered": "Move to...",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Currently Reading",
              "value": "currentlyReading",
            },
            "ref": null,
            "rendered": "Currently Reading",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Want to Read",
              "value": "wantToRead",
            },
            "ref": null,
            "rendered": "Want to Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Read",
              "value": "read",
            },
            "ref": null,
            "rendered": "Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "None",
              "value": "none",
            },
            "ref": null,
            "rendered": "None",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Testing BookShelfChanger Component render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookShelfChanger
    book={Object {}}
    onUpdateBook={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value={undefined}>
        <option
                disabled={true}
                value="">
                Move to...
        </option>
        <option
                value="currentlyReading">
                Currently Reading
        </option>
        <option
                value="wantToRead">
                Want to Read
        </option>
        <option
                value="read">
                Read
        </option>
        <option
                value="none">
                None
        </option>
</select>,
      "className": "book-shelf-changer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            disabled={true}
            value="">
            Move to...
</option>,
          <option
            value="currentlyReading">
            Currently Reading
</option>,
          <option
            value="wantToRead">
            Want to Read
</option>,
          <option
            value="read">
            Read
</option>,
          <option
            value="none">
            None
</option>,
        ],
        "onChange": [Function],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Move to...",
            "disabled": true,
            "value": "",
          },
          "ref": null,
          "rendered": "Move to...",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Currently Reading",
            "value": "currentlyReading",
          },
          "ref": null,
          "rendered": "Currently Reading",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Want to Read",
            "value": "wantToRead",
          },
          "ref": null,
          "rendered": "Want to Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Read",
            "value": "read",
          },
          "ref": null,
          "rendered": "Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "None",
            "value": "none",
          },
          "ref": null,
          "rendered": "None",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value={undefined}>
          <option
                    disabled={true}
                    value="">
                    Move to...
          </option>
          <option
                    value="currentlyReading">
                    Currently Reading
          </option>
          <option
                    value="wantToRead">
                    Want to Read
          </option>
          <option
                    value="read">
                    Read
          </option>
          <option
                    value="none">
                    None
          </option>
</select>,
        "className": "book-shelf-changer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              value="">
              Move to...
</option>,
            <option
              value="currentlyReading">
              Currently Reading
</option>,
            <option
              value="wantToRead">
              Want to Read
</option>,
            <option
              value="read">
              Read
</option>,
            <option
              value="none">
              None
</option>,
          ],
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Move to...",
              "disabled": true,
              "value": "",
            },
            "ref": null,
            "rendered": "Move to...",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Currently Reading",
              "value": "currentlyReading",
            },
            "ref": null,
            "rendered": "Currently Reading",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Want to Read",
              "value": "wantToRead",
            },
            "ref": null,
            "rendered": "Want to Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Read",
              "value": "read",
            },
            "ref": null,
            "rendered": "Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "None",
              "value": "none",
            },
            "ref": null,
            "rendered": "None",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
